apps/system/js/utility_tray.js

```diff
@@ -396,6 +428,17 @@ window.UtilityTray = {
       var significant = (Math.abs(this.lastDelta) > (this.screenHeight / 5));
       var shouldOpen = significant ? !this.shown : this.shown;
 
+      if (significant && this.lastTime <= this.VELOCITY_THRESHOLD &&
```

Comment:
I'm not sure I understand what the velocity threshold does. It seems like the spirit is to prevent a really slow drag from becoming a really slow animation? In that case, would it be better to just calculate the final velocity, and use a velocity threshold based on px/ms rather than just ms? Right now, with how `this.lastTime` gets updated (which appears to be only when the swipe is accelerating/decelerating) it's unclear what this does.